librz/core/cmd_type.c

```diff
@@ -1153,6 +1247,78 @@ static void type_define(RzCore *core, const char *type) {
 	}
 }
 
+static void types_load_file(RzCore *core, const char *path) {
+	const char *dir = rz_config_get(core->config, "dir.types");
+	char *homefile = NULL;
+	if (*path == '~') {
+		if (path[1] && path[2]) {
+			homefile = rz_str_home(path + 2);
+			path = homefile;
+		}
+	}
+	if (!strcmp(path, "-")) {
+		char *tmp = rz_core_editor(core, "*.h", "");
+		if (tmp) {
+			char *error_msg = NULL;
+			char *out = rz_parse_c_string(core->analysis, tmp, &error_msg);
+			if (out) {
+				rz_analysis_save_parsed_type(core->analysis, out);
+				free(out);
+			}
+			if (error_msg) {
+				fprintf(stderr, "%s", error_msg);
+				free(error_msg);
+			}
+			free(tmp);
+		}
+	} else {
+		char *error_msg = NULL;
+		char *out = rz_parse_c_file(core->analysis, path, dir, &error_msg);
+		if (out) {
+			rz_analysis_save_parsed_type(core->analysis, out);
+			free(out);
+		}
+		if (error_msg) {
+			fprintf(stderr, "%s", error_msg);
+			free(error_msg);
+		}
+	}
+	free(homefile);
+}
+
+static void types_load_editor(RzCore *core, const char *typename) {
+	Sdb *TDB = core->analysis->sdb_types;
+	char *str = rz_core_cmd_strf(core, "tc %s", typename);
```

Comment:
`typename` might be null when called from the newshell handler, because the argument is optional. I think this is not alright.